src/main/kotlin/ar/edu/unahur/obj2/vendedores/CentroDistribucion.kt

```diff
@@ -0,0 +1,25 @@
+package ar.edu.unahur.obj2.vendedores
+
+abstract class CentroDistribucion {
+    val vendedores = mutableListOf<Vendedor>()
+    val genericos = mutableListOf<Vendedor>()
+
+    fun agregarVendedoresNuevoYGenericoOError(vendedor: Vendedor) {
+        if (vendedores.contains(vendedor)) {
+            throw Exception("el vendedor ya esta en el centro")
+        }
+        else {
+            this.agregarVendedor(vendedor)
+            this.agregarSiEsGenerico(vendedor)
+        }
+    }
```

Comment:
El `else` no es necesario en el caso que entre por el `if` se va a lanzar la excepción y ya no se va a ejecutar la parte de agregar. Tené en cuenta también el comentario que te dejé abajo.
```suggestion
    fun agregarVendedor(vendedor: Vendedor) {
        if (vendedores.contains(vendedor)) {
            throw Exception("el vendedor ya esta en el centro")
        }
        this.agregarVendedor(vendedor)
    }
```